service/application_windows.go

```diff
@@ -120,28 +120,69 @@ func openEventLog(serviceName string) (*eventlog.Log, error) {
 	return elog, nil
 }
 
-func newWithEventViewerLoggingHook(params Parameters, elog *eventlog.Log) (*Application, error) {
+func newWithEventViewerWithWindowsCore(params Parameters, elog *eventlog.Log) (*Application, error) {
 	params.LoggingOptions = append(
 		params.LoggingOptions,
-		zap.Hooks(func(entry zapcore.Entry) error {
-			msg := fmt.Sprintf("%v\r\n\r\nStack Trace:\r\n%v", entry.Message, entry.Stack)
-
-			switch entry.Level {
-			case zapcore.FatalLevel, zapcore.PanicLevel, zapcore.DPanicLevel:
-				// golang.org/x/sys/windows/svc/eventlog does not support Critical level event logs
-				return elog.Error(3, msg)
-			case zapcore.ErrorLevel:
-				return elog.Error(3, msg)
-			case zapcore.WarnLevel:
-				return elog.Warning(2, msg)
-			case zapcore.InfoLevel:
-				return elog.Info(1, msg)
-			}
-
-			// ignore Debug level logs
-			return nil
-		}),
+		zap.WrapCore(withWindowsCore(elog)),
 	)
-
 	return New(params)
 }
+
+var _ zapcore.Core = (*windowsCore)(nil)
+
+type windowsCore struct {
```

Comment:
```suggestion
// winEvtLogCore outputs logs to Windows Event Log API.
type winEvtLogCore struct {
```

Reply:
Good call, updated to `windowsEventLogCore` as well as invoking function.